src/main/kotlin/cacophony/codegen/instructions/matching/InstructionMatcher.kt

```diff
@@ -53,10 +55,26 @@ class InstructionMatcherImpl(
                 )
             }
         }
-    }
 
-    override fun findMatchesForCondition(node: CFGNode, destinationLabel: BlockLabel, jumpIf: Boolean): Set<Match> {
-        return findMatchesWithInstructionMaker(
+    override fun findMatchesWithoutTemporaryRegisters(node: CFGNode): Set<Match> =
+        findMatchesWithInstructionMaker(
+            node,
+            noTemporaryRegistersPatterns,
+        ) { metadata: MatchMetadata, pattern: SideEffectPattern ->
+            { mapping: ValueSlotMapping ->
+                pattern.makeInstance(
+                    SlotFill(
```

Comment:
can we use this opportunity to extract this SlotFill(...) into a separate function? It's quite repetetive now